extensions/amp-auto-lightbox/0.1/carousel-criteria.js

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Layout} from '../../../src/layout';
+import {Services} from '../../../src/services';
+import {dev, devAssert} from '../../../src/log';
+import {isAmpElement, iterateCursor} from '../../../src/dom';
+import {isExperimentOn} from '../../../src/experiments';
+import {resolveFalse, resolveTrue} from './utils/promise';
+import {toArray, toWin} from '../../../src/types';
+import {tryResolve} from '../../../src/utils/promise';
+
+const MAX_TRAVERSING_DEPTH = 8;
+const MAX_TRAVERSING_BREADTH = 8;
+
+const MIN_IMG_SLIDE_AREA_RATIO = 0.5;
+
+/** @private @const {!Object<string, boolean>} */
+const ADMISSIBLE_AMP_ELEMENTS = {
```

Comment:
As long as these aren't `export`ed, they can be wrapped in a `dict`. This will also prevent their property names from from being mangled